Source/JavaScriptCore/builtins/RegExpPrototype.js

```diff
@@ -404,7 +404,8 @@ function search(strArg)
     // Check for observable side effects and call the fast path if there aren't any.
     if (@isRegExpObject(regexp)
         && @tryGetById(regexp, "exec") === @regExpBuiltinExec
-        && typeof regexp.lastIndex === "number")
+        && (typeof regexp.lastIndex === "number")
+        && @regExpLastIndexIsWritable.@call(regexp))
```

Comment:
We should make it non function call and optimize this via some dfg nodes / watchpoint since this is very hot path.